src/rgw/rgw_sal_rados.cc

```diff
@@ -538,8 +538,20 @@ int RadosBucket::set_acl(const DoutPrefixProvider* dpp, RGWAccessControlPolicy &
 
   acls = acl;
   acl.encode(aclbl);
+  map<string, bufferlist> attrs = get_attrs();
```

Comment:
`get_attrs()` is returning a reference to `sal::Bucket::attrs`. i think we should be modifying that, instead of making a copy here first. so this `attrs` variable should a reference

that way, after calling `set_acl()`, another call to `get_attrs()` would see this updated `RGW_ATTR_ACL`

Reply:
got it.. thanks! done.